test/tests/selftest/pbs_pbstestsuite.py

```diff
@@ -0,0 +1,222 @@
+        self.assertEqual(pbs_conf_val["PBS_START_SCHED"], "1")
+        self.assertEqual(pbs_conf_val["PBS_START_COMM"], "1")
+
+        # Check that the remote mom's pbs.conf has mom start bit on
+        pbs_conf_filepath = self.du.get_pbs_conf_file(remotemom.hostname)
+        pbs_conf_val = self.du.parse_pbs_config(remotemom.hostname,
+                                                pbs_conf_filepath)
+        self.assertEqual(pbs_conf_val["PBS_START_MOM"], "1")
+
+        # Now set it to 0 and restart the mom
+        remotemom.pi.stop(remotemom.hostname)
+        pbs_conf_val["PBS_START_MOM"] = "0"
+        self.du.set_pbs_config(fin=pbs_conf_filepath, confs=pbs_conf_val)
+        remotemom.pi.start(remotemom.hostname)
+
+        # Confirm that the mom is down
+        self.assertFalse(remotemom.isUp())
+
+        # Now call revert_pbsconf() & restart mom again
+        self.revert_pbsconf()
+
+        # Confirm that the mom came up and start bit is 1
+        self.assertTrue(remotemom.isUp())
+        pbs_conf_val = self.du.parse_pbs_config(remotemom.hostname)
+        self.assertEqual(pbs_conf_val["PBS_START_MOM"], "1")
+
+    def test_revert_pbsconf_corelimit(self):
+        """
+        Test the functionality of PBSTestSuite.revert_pbsconf() when
+        PBS_CORE_LIMIT is reverted
+        """
+        pbs_conf_filepath = self.du.get_pbs_conf_file(self.server.hostname)
+        pbs_conf_val = self.du.parse_pbs_config(self.server.hostname,
+                                                pbs_conf_filepath)
+        if pbs_conf_val is None or len(pbs_conf_val) < 1:
+            raise ValueError("Could not parse pbs.conf on host %s" %
+                             (server.hostname))
+
+        # Since the setUp already ran, check that PBS_CORE_LIMIT is set to
+        # unlimited
+        self.assertEqual(pbs_conf_val["PBS_CORE_LIMIT"], "unlimited")
+
+        # Now, set the core limit to 0 and restart PBS
+        self.server.pi.stop()
+        pbs_conf_val["PBS_CORE_LIMIT"] = "0"
+        self.du.set_pbs_config(fin=pbs_conf_filepath, confs=pbs_conf_val)
+        self.server.pi.start()
+
+        # First, check that there's no existing core file in mom_priv
+        parent_dir = os.getcwd()
+        snapdirs = []
+        with PBSSnapUtils(out_dir=parent_dir, acct_logs=0, daemon_logs=0,
+                          with_sudo=True) as snap_obj:
+            snap_dir = snap_obj.capture_nodes(with_mom_logs=False)
+            snapdirs.append(snap_dir)
+            core_dir_path = os.path.join(snap_dir, CORE_MOM_PATH)
+            if os.path.isdir(core_dir_path):
+                # Delete all existing core files
+                core_filenames = os.listdir(core_dir_path)
+                for corefile in core_filenames:
+                    pbs_corepath = os.path.join(pbs_conf_val["PBS_HOME"],
+                                                "mom_priv", corefile)
+                    self.du.rm(path=pbs_corepath, force=True, sudo=True)
```

Comment:
core files are dropped in mom_priv.  This will delete everything in mom's priv directory (like her config file).  Also, I don't think we need to test this.  We aren't testing the mom functionality of dropping a core or not.  We are testing whether or not we revert pbs.conf.

Reply:
Well, it does test that calling revert_pbsconf() makes the changes take effect in daemons. If we do not add this test then we don't know if daemons have been restarted and setup has correctly reverted it.

Reply:
Fair enough.  I can see the logic in that.  How do we know if the daemon has been restarted if we don't test the functionality that has been reverted.  The first part of my comment is still true though.  There are other files in mom_priv other than core files.  We should not delete them.

Reply:
We aren't deleting anything other than the core files. core_filenames literally is a list of just the core files, and we only delete them from mom_priv.

Reply:
Isn't core_filenames a listdir() of mom_priv?

Reply:
no, it is a listdir on "core_file_bt/mom_priv", that directory only contains backtraces from core files. I'm going to take Arun's advice and just look for files which start with 'core' instead of taking a snapshot, that should make the test look much more straight-forward